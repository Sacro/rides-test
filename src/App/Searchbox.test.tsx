import 'jest-enzyme';

import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import SearchBox from './Searchbox';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  shallow(<SearchBox />);
});

it('renders correctly', () => {
  const SearchboxComponent = mount(<SearchBox />);
  expect(SearchboxComponent).toMatchSnapshot();
});

describe('Acceptance Tests', () => {
  describe('AC1', () => {
    it('should have a Pick-up Location widget', () => {});
  });

  describe('AC2', () => {
    it('should have the correct placeholder text', () => {
      const SearchboxComponent = mount(<SearchBox />);
      expect(SearchboxComponent.find('input').props().placeholder).toBe(
        'city, airport, station, region and district...'
      );
    });
    1;
  });

  /*describe('AC3', () => {
    it('should apply a focus state', () => {
      const SearchboxComponent = mount(<SearchBox />);
      const input = SearchboxComponent.find('input');
      expect(input.props().frameBorder).toBe(3);
    });
  }); //*/
});
