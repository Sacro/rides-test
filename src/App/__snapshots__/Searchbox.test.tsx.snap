// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Searchbox>
  <div
    style={
      Object {
        "backgroundColor": "#F3CE56",
        "padding": "1em",
      }
    }
  >
    <h1
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
    >
      Where are you going?
    </h1>
    <Form
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
      prefixCls="ant-form"
    >
      <form
        className="ant-form ant-form-horizontal"
        onSubmit={[Function]}
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Row
            className="ant-form-item"
            gutter={0}
          >
            <div
              className="ant-row ant-form-item"
            >
              <Col
                className="ant-form-item-control-wrapper"
                key="wrapper"
              >
                <div
                  className="ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    >
                      <label
                        style={
                          Object {
                            "float": "left",
                            "textAlign": "left",
                          }
                        }
                      >
                        Pick-up Location?
                      </label>
                      <Input
                        className="location"
                        disabled={false}
                        placeholder="city, airport, station, region and district..."
                        prefixCls="ant-input"
                        type="text"
                      >
                        <input
                          className="ant-input location"
                          disabled={false}
                          onKeyDown={[Function]}
                          placeholder="city, airport, station, region and district..."
                          type="text"
                        />
                      </Input>
                    </span>
                    <Animate
                      animation={Object {}}
                      component=""
                      componentProps={Object {}}
                      key="help"
                      onAppear={[Function]}
                      onEnd={[Function]}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="show-help"
                    />
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </FormItem>
      </form>
    </Form>
  </div>
</Searchbox>
`;
